SurfaceDevCenterManager/DevCenterApi/DevCenterHandler.cs

```diff
@@ -101,25 +106,46 @@ private string GetDevCenterBaseUrl()
                         }
                     }
 
-                    break;
-                }
+                    string content = await infoResult.Content.ReadAsStringAsync();
+                    if (infoResult.IsSuccessStatusCode)
+                    {
+                        processContent?.Invoke(content);
+                        return null;
+                    }
 
-                string content = await infoResult.Content.ReadAsStringAsync();
-                if (infoResult.IsSuccessStatusCode)
-                {
-                    processContent?.Invoke(content);
-                    return null;
-                }
+                    try
+                    {
+                        reterr = JsonConvert.DeserializeObject<DevCenterErrorReturn>(content);
+                    }
+                    catch (JsonReaderException)
+                    {
+                        // Error is in bad format, return raw
+                        reterr = new DevCenterErrorReturn()
+                        {
+                            HttpErrorCode = (int)infoResult.StatusCode,
+                            StatusCode = infoResult.StatusCode.ToString("D") + " " + infoResult.StatusCode.ToString(),
+                            Message = content
+                        };
+                    }
 
-                DevCenterErrorReturn reterr = JsonConvert.DeserializeObject<DevCenterErrorReturn>(content);
-                // reterr can be null when there is HTTP error
-                if (reterr == null)
-                {
-                    return new DevCenterErrorDetails
+                    // reterr can be null when there is HTTP error
+                    if (reterr == null)
                     {
-                        Code = infoResult.StatusCode.ToString("D"),
-                        Message = infoResult.ReasonPhrase
-                    };
+                        reterr = new DevCenterErrorReturn()
+                        {
+                            HttpErrorCode = (int)infoResult.StatusCode,
+                            StatusCode = infoResult.StatusCode.ToString("D") + " " + infoResult.StatusCode.ToString(),
+                            Message = infoResult.ReasonPhrase
+                        };
+                    }
+
+                    // Retry on the following 'infrastructure' issues
+                    // 502 BadGateway
+                    if (infoResult.StatusCode != System.Net.HttpStatusCode.BadGateway)
+                    {
+                        break;
+                    }
+                    await Task.Delay(new Random().Next(1000, 10000));
                 }
 
```

Comment:
Does the return path at line 153 need to return a 'DevCenterErrorDetails'? The HttpErrorCode is not being set here. 
_(I couldn't add a comment to unchanged code, so adding it here)_